### PR TITLE
Implement dashboard screen

### PR DESCRIPTION
**Pull request by ryonryon**
@axross Can you check these changes?

### CODE REVIEW

File: lib/src/widgets/dashboard_screen/cards_maturity_donut_chart.dart [hunk 1]
```diff
@@ -0,0 +1,54 @@
+import 'package:flutter/material.dart';
+import 'package:charts_flutter/flutter.dart' as charts;
+
+class CardsMaturityDonutChart extends StatelessWidget {
+  final List<charts.Series> seriesList;
+  final bool animate;
+
+  CardsMaturityDonutChart(this.seriesList, {this.animate});
```
Comment (axross) [inline lib/src/widgets/dashboard_screen/cards_maturity_donut_chart.dart]: It seems not necessary to bypass implementation via `withData` factory constructor and `_createData` static method. You should organize them and make ones to be simple.

File: lib/src/widgets/dashboard_screen/cards_maturity_donut_chart.dart [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    ];
+  }
+}
+
+class MatureCount {
+  final String title;
+  final int maturity;
+
+  MatureCount(this.title, this.maturity);
+}
```
Comment (axross) [inline lib/src/widgets/dashboard_screen/cards_maturity_donut_chart.dart]: There's supposed to be blank line in the end of file. You must format code with dartfmt. Note that there's also several dirty code else.

File: lib/src/widgets/dashboard_screen/cards_maturity_donut_chart.dart [hunk 3]
```diff
@@ -0,0 +1,54 @@
+import 'package:flutter/material.dart';
+import 'package:charts_flutter/flutter.dart' as charts;
+
+class CardsMaturityDonutChart extends StatelessWidget {
```
Comment (axross) [inline lib/src/widgets/dashboard_screen/cards_maturity_donut_chart.dart]: Supposed to be `CardMaturityDonutChart` (not plural in this case)

File: lib/src/widgets/dashboard_screen/dashboard_gridview_item.dart [hunk 4]
```diff
@@ -0,0 +1,124 @@
+import 'package:flutter/material.dart';
+import 'package:flutter/widgets.dart';
+import './cards_maturity_donut_chart.dart';
+import 'package:saraka/constants.dart';
+
+class CardsMaturityGridViewItem extends StatelessWidget {
```
Comment (axross) [inline lib/src/widgets/dashboard_screen/dashboard_gridview_item.dart]: The file name and class name is not match.

Comment (axross) [reply lib/src/widgets/dashboard_screen/dashboard_gridview_item.dart]: If anything, this widget name doesn't make sense. How about `Summary` instead?

File: lib/src/widgets/dashboard_screen/dashboard_gridview_item.dart [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+                      Text(
+                        totalCardsNumber,
+                        overflow: TextOverflow.ellipsis,
+                        style: const TextStyle(
+                          color: SarakaColors.lightBlack,
+                          fontSize: 30.0,
+                          fontFamily: SarakaFonts.rubik,
+                          fontWeight: FontWeight.w500,
+                          height: 1.25,
+                        ),
```
Comment (axross) [inline lib/src/widgets/dashboard_screen/dashboard_gridview_item.dart]: Use `SarakaTextStyles.xxx` as much as possible. See #75.

File: lib/src/widgets/dashboard_screen/dashboard_gridview_item.dart [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+                          Text(
+                            'Cards',
+                            overflow: TextOverflow.ellipsis,
+                            style: const TextStyle(
+                              color: SarakaColors.lightBlack,
+                              fontSize: 12.0,
+                              fontFamily: SarakaFonts.rubik,
+                              fontWeight: FontWeight.w500,
+                              height: 1.25,
+                            ),
```
Comment (axross) [inline lib/src/widgets/dashboard_screen/dashboard_gridview_item.dart]: Use `SarakaTextStyles.xxx` as much as possible. See #75.

File: lib/src/widgets/dashboard_screen/dashboard_gridview_item.dart [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+                              Text(
+                                'See card list',
+                                overflow: TextOverflow.ellipsis,
+                                style: const TextStyle(
+                                  color: SarakaColors.lightBlack,
+                                  fontSize: 20.0,
+                                  fontFamily: SarakaFonts.rubik,
+                                  fontWeight: FontWeight.w500,
+                                  height: 1.25,
+                                ),
```
Comment (axross) [inline lib/src/widgets/dashboard_screen/dashboard_gridview_item.dart]: Use `SarakaTextStyles.xxx` as much as possible. See #75.

File: lib/src/widgets/dashboard_screen/dashboard_gridview_item.dart [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+                      )
+                    ],
+                  )),
+                ],
+              ),
+            ),
+            Container(
+              child: Center(
+                child: Text(
+                  todayLearnNumber + ' cards you study for today',
```
Comment (axross) [inline lib/src/widgets/dashboard_screen/dashboard_gridview_item.dart]: I think `X cards to study today` is better 🤔

File: lib/src/widgets/dashboard_screen/dashboard_gridview_item.dart [hunk 9]
```diff
@@ -0,0 +1,126 @@
...
+                  overflow: TextOverflow.ellipsis,
+                  style: const TextStyle(
+                    color: SarakaColors.lightBlack,
+                    fontSize: 20.0,
+                    fontFamily: SarakaFonts.rubik,
+                    fontWeight: FontWeight.w500,
+                    height: 1.25,
+                  ),
```
Comment (axross) [inline lib/src/widgets/dashboard_screen/dashboard_gridview_item.dart]: Use SarakaTextStyles.xxx as much as possible. See #75.

File: lib/src/widgets/dashboard_screen/main_drawer.dart [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+    super.initState();
+
+    Future.delayed(Duration.zero, () {
+      final authenticationBloc = Provider.of<AuthenticationBloc>(context);
+
+      setState(() {
+        _authenticationBloc = authenticationBloc;
+      });
+    });
+  }
```
Comment (axross) [inline lib/src/widgets/dashboard_screen/main_drawer.dart]: You don't have to choose to use `StatefulWidget` and `initState` for this case. You can use a `Consumer` widget in `build()` to retrieve a bloc from the ancestor.

File: lib/src/widgets/dashboard_screen/main_drawer.dart [hunk 11]
```diff
@@ -0,0 +1,89 @@
...
+      ),
+    );
+  }
+
+  void _onTap() {
+    final authenticationBloc = Provider.of<AuthenticationBloc>(context);
+
+    authenticationBloc.signOut();
+  }
+}
```
Comment (axross) [inline lib/src/widgets/dashboard_screen/main_drawer.dart:69]: There's still `MainDrawer` in `StudyScreen`. It's supposed to be deleted (We can make it into an issue and fix it in another pull request)

File: lib/src/widgets/dashboard_screen/start_learning_floating_action_button.dart [hunk 12]
```diff
@@ -0,0 +1,21 @@
...
+        icon: Icon(Feather.getIconData('play')),
+        label: Text(
+          'Start learning',
+          style: TextStyle(
+            fontFamily: SarakaFonts.rubik,
+            fontSize: 14,
+          ),
+        ),
+        shape: SuperellipseShape(borderRadius: BorderRadius.circular(24)),
+        onPressed: () => Navigator.of(context).pushNamed('/study'),
```
Comment (axross) [inline lib/src/widgets/dashboard_screen/start_learning_floating_action_button.dart:16]: :+1:

File: lib/src/widgets/study_screen/study_screen.dart [hunk 13]
```diff
@@ -47,16 +47,13 @@ class StudyScreen extends StatelessWidget {
...
-                  onPressed: () => Navigator.of(context).pushNamed('/cards'),
-                ),
-              ],
-              leading: FeatherDrawerButton(),
+              leading: Navigator.of(context).canPop()
+                  ? IconButton(
+                      icon: Icon(Feather.getIconData('arrow-left')),
+                      onPressed: () => Navigator.of(context).pop(),
+                    )
+                  : null,
```
Comment (axross) [inline lib/src/widgets/study_screen/study_screen.dart]: :+1:

**Review by axross — CHANGES_REQUESTED** [review]
Thanks for great work! On the other hand I left several comments. I want you to fix them before merging.

File: lib/src/widgets/dashboard_screen/cards_maturity_donut_chart.dart (hunk 1 above)
Comment (ryonryon) [reply lib/src/widgets/dashboard_screen/cards_maturity_donut_chart.dart]: Done

File: lib/src/widgets/dashboard_screen/cards_maturity_donut_chart.dart (hunk 2 above)
Comment (ryonryon) [reply lib/src/widgets/dashboard_screen/cards_maturity_donut_chart.dart]: Done

File: lib/src/widgets/dashboard_screen/cards_maturity_donut_chart.dart (hunk 3 above)
Comment (ryonryon) [reply lib/src/widgets/dashboard_screen/cards_maturity_donut_chart.dart]: Done

File: lib/src/widgets/dashboard_screen/dashboard_gridview_item.dart (hunk 4 above)
Comment (ryonryon) [reply lib/src/widgets/dashboard_screen/dashboard_gridview_item.dart]: Good, Done.

File: lib/src/widgets/dashboard_screen/main_drawer.dart (hunk 11 above)
Comment (ryonryon) [reply lib/src/widgets/dashboard_screen/main_drawer.dart:69]: OK!

File: lib/src/widgets/dashboard_screen/dashboard_gridview_item.dart (hunk 9 above)
Comment (ryonryon) [reply lib/src/widgets/dashboard_screen/dashboard_gridview_item.dart]: Done.

File: lib/src/widgets/dashboard_screen/dashboard_gridview_item.dart (hunk 5 above)
Comment (ryonryon) [reply lib/src/widgets/dashboard_screen/dashboard_gridview_item.dart]: Done.

File: lib/src/widgets/dashboard_screen/main_drawer.dart (hunk 11 above)
Comment (ryonryon) [reply lib/src/widgets/dashboard_screen/main_drawer.dart:69]: I deleted MainDrawer.

File: lib/src/widgets/dashboard_screen/dashboard_screen.dart [hunk 14]
```diff
@@ -0,0 +1,61 @@
+import 'package:flutter/material.dart' hide Card;
+import 'package:flutter/widgets.dart';
+import 'package:flutter_staggered_grid_view/flutter_staggered_grid_view.dart';
+import 'package:saraka/constants.dart';
+import 'package:provider/provider.dart';
+import 'package:saraka/blocs.dart';
+import './summary.dart';
+import './main_drawer.dart';
+import './start_learning_floating_action_button.dart';
+import './wave_background.dart';
```
Comment (axross) [inline lib/src/widgets/dashboard_screen/dashboard_screen.dart:10]: Can you make them in the order? (a to z)

**Review by axross — APPROVED** [review]
I left just one comment but mostly looks good to me!

File: lib/src/widgets/dashboard_screen/dashboard_screen.dart (hunk 14 above)
Comment (ryonryon) [reply lib/src/widgets/dashboard_screen/dashboard_screen.dart:10]: @axross thanks.

**Pull request merged**